### DISCUSSION TITLE
Decorating the cancellation token from a behaviour

### DISCUSSION THREAD

**Discussion by dazinator** [LuckyPennySoftware/MediatR#1056] [General]
As per https://github.com/jbogard/MediatR/issues/592
It seems like a discussion was suggested, so I'd like to create on here.

**Comment by zachpainter77** [comment]
So.. what are you trying to achieve?  It seems like people are wanting to set timeouts on mediator calls from a behavior.  Is that correct?  I could take a whack at it.

Edit: I have just completed a PR adding this support.

#1058

**Reply by dazinator** [reply]
Amazing thank you. I will watch the PR in anticipation! To elaborate on my use case:-
1. I have implemented a pipeline behavior to address distributed locking as a cross-cutting concern. The behavior acquires a distributed lock before forwarding the request.
2. The behavior works nicely, but we have a few requests that are long running async operations and we saw an instance where the lock was being lost.. the behavior is able to be notified when the lock is lost, but it has no way to signal the cancellation token of the active handler which continued to process the request.

The workaround is to add a seperate cancellation token to the request, which the behaviour can set, and signal independently - but then this means there are now two cancellation tokens for the handler to check so it isn't a clean as i'd like. One cancellation token should be enough to signal the cancellation of the async operation as per mediatr's handler method signature - it's just currently, mediatr doesn't give me a way to intercept that cancellation token, so I can't for example swap it for a linked cancellation token that incorporates cancellation token sources brought in by a behaviour.

**Reply by zachpainter77** [reply]
@dazinator It looks like the timeout behavior pr was merged.. Are you able to request the handler cancellation now?

**Reply by DonnelCyril** [reply]
Unfortunately not.

I can see the change in the main branch but not in the package I am consuming - version 12.4.1. 

![image](https://github.com/user-attachments/assets/385e8783-f0e0-400f-91eb-84796673d8ef)

I am assuming an explicit release is required to push this out?

The last release was 2 days before this got merged :-( 
![image](https://github.com/user-attachments/assets/c395ff34-30f0-45fd-8cf3-36e9e5f9bf66)
 
 What is the process to request a release that contains this change?
